new-cdc/eventuate-local-java-cdc-connector-unified/src/main/java/io/eventuate/local/unified/cdc/pipeline/CdcPipelineConfiguration.java

```diff
@@ -93,24 +92,20 @@ public void stop() {
     cdcPipelines.forEach(CdcPipeline::stop);
   }
 
-  private List<Map<String, Object>> convertPropertiesToListOfMaps(String properties) {
-    try {
-      return objectMapper.readValue(properties, List.class);
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-  }
+  private void createStartSaveCdcPipeline(Map<String, Object> properties) {
+    propertyReader.checkForUnknownProperties(properties, CdcPipelineProperties.class);
+
+    CdcPipelineProperties cdcPipelineProperties = propertyReader
+            .convertMapToPropertyClass(properties, CdcPipelineProperties.class);
```

Comment:
Why does convertMapToPropertyClass() call checkForUnknownProperties()?

Reply:
Maybe you meant 'doesn't'?
If so, because sometimes it is not desirable, this method is also used in the reader properties creation. And to find exact property class, properties converted to the common superclass first (it does not know about most properties), which has type of the factory, which knows exact property class it should be.